src/lib.rs

```diff
@@ -120,9 +137,9 @@ where
     pub fn set_datarate(&mut self, datarate: DataRate) -> Result<(), Error<E>> {
```

Comment:
The data rate and mode are intermingled for certain data rates.

The easiest fix is to just remove 1.6, 1.344 and 5.376 rates because they cant actually be set independently from a mode. Or we could check and error in the data rate (and mode!) setters.

Or we could have mode and rate be set with the same function, maybe even the same enum like HR1hz, Low1hz, normal1hz, low5.376hz.  Or maybe an low(1hz), low(5.376hz)

8.8 Table 31
ODR
0000 Power-down mode
0001 HR / Normal / Low-power mode (1 Hz)
0010 HR / Normal / Low-power mode (10 Hz)
0011 HR / Normal / Low-power mode (25 Hz)
0100 HR / Normal / Low-power mode (50 Hz)
0101 HR / Normal / Low-power mode (100 Hz)
0110 HR / Normal / Low-power mode (200 Hz)
0111 HR / Normal / Low-power mode (400 Hz)
1000 Low power mode (1.60 kHz)
1001HR / normal (1.344 kHz);Low-power mode (5.376 kHz)